File: Easy/JavaScript/349. Intersection of Two Arrays.js
var intersection = function(nums1, nums2) {
    let freq = new Set();

    for(let i = 0; i < nums1.length; i++){
        for(let j = 0; j < nums2.length; j++){
            if(nums1[i] == nums2[j] && !freq.has(nums1[i])){
                freq.add(nums1[i]);
                break;
            }
        }
    }

    let count = Array(freq.size).fill(0);
    let index = 0;
    for(let n of freq){
        count[index] = n;
        index++;
    }

    return count;
};

File: Easy/Java/349. Intersection of Two Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        int[] freq = new int[1001];

        for(int n : nums1){
            if(freq[n] == 0)
                freq[n]++;
        }

        int count = 0;
        for(int n : nums2){
            if(freq[n] == 1){
                freq[n]++;
                count++;
            }
        }

        int[] ans = new int[count];
        int index = 0;
        
        for(int i = 0; i < freq.length; i++){
            if(freq[i] == 2){
                ans[index] = i;
                index++;
            }
        }
        
        return ans;
    }
}
